src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/SearchController.cs

```diff
@@ -75,6 +75,11 @@ public async Task<IActionResult> Search(SearchIndexViewModel viewModel, PagerSli
             var siteSettings = await _siteService.GetSiteSettingsAsync();
             var searchSettings = siteSettings.As<LuceneSettings>();
 
+            if(!String.IsNullOrWhiteSpace(viewModel.Index))
+            {
+                searchSettings.SearchIndex = viewModel.Index;
```

Comment:
As mentioned by the below summary, here `searchSettings` should not be mutated, you can use a separate variable initialized with `searchSettings.SearchIndex` and that may be overridden by `viewModel.Index`, and then use it.

https://github.com/OrchardCMS/OrchardCore/blob/7c3d8019f77e6f27e87b754ae78f910216c017df/src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/ISiteService.cs#L15-L18